warehouse/integrations/bigquery/bigquery.go

```diff
@@ -867,7 +867,7 @@ func (bq *BigQuery) connect(ctx context.Context, cred BQCredentials) (*bigquery.
 // * the server config says we allow merging
 // * the user opted in to merging
 func (bq *BigQuery) shouldMerge() bool {
-	return bq.config.allowMerge && bq.warehouse.GetBoolDestinationConfig(model.EnableMergeSetting)
+	return bq.config.allowMerge && bq.warehouse.GetEnableMergeSetting()
```

Comment:
```suggestion
	return bq.config.allowMerge && !bq.warehouse.PreferAppend()
```

What if we inverse our logic here. Instead of enableMerge, we change the option to preferAppend false by default, customers can opt-in from UI.

Reply:
Yes if we want to existing customer to use Merge then no migration is required

Reply:
> if it is null in the config, what will happen if you just edit and save without changing anything ?

in this case `enableMerge` would become `false` making it more risky than `preferAppend`. will rename the field to `preferAppend` to make it more robust.